packaging/windows/WindowsLauncher.cs.in

```diff
@@ -12,104 +12,81 @@
 using System;
 using System.Diagnostics;
 using System.Drawing;
-using System.IO;
 using System.Linq;
 using System.Media;
 using System.Reflection;
-using System.Runtime.InteropServices;
 using System.Windows.Forms;
 
 namespace OpenRA
 {
 	class WindowsLauncher
 	{
-		[DllImport("user32.dll")]
-		static extern int SendMessage(IntPtr hwnd, uint message, uint wParam, IntPtr lParam);
+		static Process _gameProcess;
 
-		[DllImport("shell32.dll")]
-		static extern IntPtr SHGetFileInfo(string pszPath, uint dwFileAttributes, ref SHFILEINFO psfi, uint cbSizeFileInfo, uint uFlags);
-
-		[DllImport("user32.dll")]
-		public extern static bool DestroyIcon(IntPtr handle);
+		// Constants to be replaced by the wrapper / compilation script
+		const string ModId = "MOD_ID";
+		const string DisplayName = "DISPLAY_NAME";
+		const string FaqUrl = "FAQ_URL";
 
-		struct SHFILEINFO
+		[STAThread]
+		static int Main(string[] args)
 		{
-			// Native type: HICON
-			public IntPtr hIcon;
-
-			public int iIcon;
-			public uint dwAttributes;
-
-			// Native type: TCHAR[MAX_PATH]
-			[MarshalAs(UnmanagedType.ByValTStr, SizeConst = 260)]
-
-			public string szDisplayName;
-
-			// Native type: TCHAR[80]
-			[MarshalAs(UnmanagedType.ByValTStr, SizeConst = 80)]
+			if (args.Any(x => x.Contains("Engine.LaunchPath")))
+			{
+				var exitCode = RunGame(args); // TODO: Test various exit scenarios.
```

Comment:
This will need to be finished before merging.

Reply:
Actually upon further inspection I should already have all the error handling covered inside `RunGame`, so there is likely no point in even returning an exit code here.